core/modules/qmeta/QMetaMysql.cc

```diff
@@ -817,6 +817,129 @@ QMetaMysql::_checkDb() {
                                value);
     }
 
+    createQueryStatsTmpTable();
+}
+
+
+void QMetaMysql::createQueryStatsTmpTable() {
+    // If this is being run outside the constructor, _bdMutex may need to be locked first.
+    // Try to create the temporary table if it is not there.
+    QMetaTransaction trans(_conn);
+    sql::SqlErrorObject errObj;
+    sql::SqlResults results;
+    std::string query = "CREATE TABLE IF NOT EXISTS QStatsTmp (queryId bigint(20), "
+                        "totalChunks int, completedChunks int, "
+                        "queryBegin timestamp DEFAULT 0, "
+                        "lastUpdate timestamp DEFAULT 0, "
+                        "PRIMARY KEY (queryId)) "
+                        "ENGINE = MEMORY;";
```

Comment:
No need to add semicolon to the statements (here and all other places).

Reply:
Just a note for the future - if schema of this table ever changes you'll also need to DROP old one.

Reply:
If (probably when) a schema change becomes needed, DROP'ing the old table should probably be done with a script. All of the data is temporary in nature, so there's no need to worry about keeping it if qserv is stopped.